cvat/apps/engine/models.py

```diff
@@ -520,6 +519,24 @@ def __str__(self):
     def has_parent_label(self):
         return bool(self.parent)
 
+    def save(self, force_insert=False, force_update=False, using=None,
+             update_fields=None):
+        try:
+            super().save(force_insert, force_update, using, update_fields)
+        except ValueError as exc:
+            raise exceptions.ValidationError(str(exc)) from exc
+        except IntegrityError:
+            raise exceptions.ValidationError("All label names must be unique")
```

Comment:
```suggestion
    def save(self, force_insert=False, force_update=False, using=None,
             update_fields=None):
        try:
            super().save(force_insert, force_update, using, update_fields)
        except IntegrityError as exc:
            raise InvalidLabel("All label names must be unique") from exc
```

I suggest to change both methods this way:
- This will encapsulate model logic about constraints in this class (InvalidLabel will need to be catched from the calling side)
- ValueError here means the data saved was incorrect, so it's an internal serializer error, not a user error

Reply:
Updated